plugins/commands/cloud/box/show.rb

```diff
@@ -32,40 +34,47 @@ def execute
                 help: opts.help.chomp
             end
 
-            @client = VagrantPlugins::CloudCommand::Util.client_login(@env, options[:username])
-            box = argv.first.split('/', 2)
+            @client = client_login(@env, options.slice(:quiet))
+            org, box_name = argv.first.split('/', 2)
 
-            show_box(box[0], box[1], options, @client.token)
+            show_box(org, box_name, @client&.token, options.slice(:versions))
           end
 
-          def show_box(org, box_name, options, access_token)
-            username = options[:username]
-
-            server_url = VagrantPlugins::CloudCommand::Util.api_server_url
-            account = VagrantPlugins::CloudCommand::Util.account(username, access_token, server_url)
-            box = VagrantCloud::Box.new(account, box_name, nil, nil, nil, access_token)
-
-            begin
-              success = box.read(org, box_name)
+          # Display the requested box to the user
+          #
+          # @param [String] org Organization name of box
+          # @param [String] box_name Name of box
+          # @param [String] access_token User access token
+          # @param [Hash] options Options for box filtering
+          # @option options [String] :versions Specific verisons of box
+          # @return [Integer]
+          def show_box(org, box_name, access_token, options={})
+            account = VagrantCloud::Account.new(
+              custom_server: api_server_url,
+              access_token: access_token
+            )
+            with_box(account: account, org: org, box: box_name) do |box|
+              if box && options[:versions]
```

Comment:
I think this should also check if versions is empty. It seems like usually the `versions` options passed as an empty array, for example running the command `vagrant cloud box show org/box`. 

Currently, this results in no boxes getting shown. Running `vagrant cloud box show org/box --versions 1.0.0` works. This suggestion resolves this

```suggestion
              if box && !Array(options[:versions]).empty?
```